controllers/yawol-controller/loadbalancermachine/loadbalancermachine_controller.go

```diff
@@ -726,55 +804,76 @@ func (r *LoadBalancerMachineReconciler) deleteServer(
 func (r *LoadBalancerMachineReconciler) deletePort(
 	ctx context.Context,
 	osClient os.Client,
-	req ctrl.Request,
 	lbm *yawolv1beta1.LoadBalancerMachine,
-) error {
+) (bool, error) {
 	var err error
+	var requeue bool
 	var portClient os.PortClient
 	portClient, err = osClient.PortClient(ctx)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	if lbm.Status.PortID != nil {
-		if err = openstackhelper.DeletePort(ctx, portClient, *lbm.Status.PortID); err != nil {
+	if lbm.Status.DefaultPortID != nil {
+		if err = openstackhelper.DeletePort(ctx, portClient, *lbm.Status.DefaultPortID); err != nil {
 			switch err.(type) {
 			case gophercloud.ErrDefault404:
-				r.Log.Info("error deleting port, already deleted", "lbm", lbm.Name)
+				_ = helper.RemoveFromLBMStatus(ctx, r.Status(), lbm, "defaultPortIP")
+				if err := helper.RemoveFromLBMStatus(ctx, r.Client.Status(), lbm, "defaultPortID"); err != nil {
+					return false, err
+				}
 			default:
 				r.Log.Info("an unexpected error occurred deleting port", "lbm", lbm.Name)
-				return kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
+				return false, kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
 			}
 		}
+		requeue = true
 	}
 
-	// delete orphan ports
-	portName := req.NamespacedName.String()
-	var portList []ports.Port
-	portList, err = portClient.List(ctx, ports.ListOpts{Name: portName})
-	if err != nil {
-		return err
-	}
-
-	for i := range portList {
-		port := &portList[i]
-		// double check
-		if port.ID == "" || port.Name != portName {
-			continue
-		}
-
-		if err = openstackhelper.DeletePort(ctx, portClient, port.ID); err != nil {
+	// TODO cleanup after removing deprecated fields
+	if lbm.Status.PortID != nil { //nolint: staticcheck // needed to be backwards compatible
+		//nolint: staticcheck // needed to be backwards compatible
+		if err = openstackhelper.DeletePort(ctx, portClient, *lbm.Status.PortID); err != nil {
 			switch err.(type) {
 			case gophercloud.ErrDefault404:
-				r.Log.Info("error deleting port, already deleted", "lbm", lbm.Name)
+				if err := helper.RemoveFromLBMStatus(ctx, r.Client.Status(), lbm, "portID"); err != nil {
+					return false, err
+				}
 			default:
 				r.Log.Info("an unexpected error occurred deleting port", "lbm", lbm.Name)
-				return kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
+				return false, kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
 			}
 		}
+		requeue = true
 	}
 
-	return nil
+	if requeue {
+		return true, nil
+	}
+
+	if lbm.Status.DefaultPortName != nil {
+		port, err := openstackhelper.GetPortByName(ctx, portClient, *lbm.Status.DefaultPortName)
```

Comment:
just wondering why this does not return a list? cant there be multiple ports with the same name and this function should clean up all of them to prevent the error we had in the past with orphan ports?

Reply:
It still deletes all ports, because it requeue after a deletion and removes the `defaultPortName` only when `openstackhelper.GetPortByName` return nil (so all ports are deleted)
I just aligned it with the lb controller.